modules/effects/src/concat_latest_from.ts

```diff
@@ -0,0 +1,80 @@
+import { Observable, of, OperatorFunction } from 'rxjs';
+import { concatMap, withLatestFrom } from 'rxjs/operators';
+
+type TypeOfObservable<T> = T extends Observable<infer U> ? U : never;
+
+/**
+ * @description
+ * RxJS operator that lazily evaluates a list of Observables.
+ *
+ * @param observablesFactory A function which returns an `Observable[]`.
+ * @returns Returns an `OperatorFunction` with ??? (idk how to describe this)
```

Comment:
deeply struggled with how to describe the return type lol. anyone, please advise